src/main/java/org/folio/search/repository/SearchRepository.java

```diff
@@ -36,8 +51,53 @@ public SearchResponse search(ResourceRequest resourceRequest, SearchSourceBuilde
       .source(searchSource)
       .indices(index);
 
-    return performExceptionalOperation(
-      () -> elasticsearchClient.search(searchRequest, RequestOptions.DEFAULT),
-      index, "searchApi");
+    return performExceptionalOperation(() -> elasticsearchClient.search(searchRequest, DEFAULT), index, "searchApi");
+  }
+
+  /**
+   * Executes scroll request to elasticsearch and transforms it to the list of instance ids.
+   *
+   * @param request resource request as {@link ResourceRequest} object.
+   * @param source elasticsearch search source as {@link SearchSourceBuilder} object.
+   */
+  public void streamResourceIds(ResourceRequest request, SearchSourceBuilder source, Consumer<List<String>> consumer) {
+    var index = getElasticsearchIndexName(request);
+    var searchRequest = new SearchRequest()
+      .scroll(new Scroll(KEEP_ALIVE_INTERVAL))
+      .routing(request.getTenantId())
+      .source(source)
+      .indices(index);
+
+    var searchResponse = performExceptionalOperation(
+      () -> elasticsearchClient.search(searchRequest, DEFAULT), index, "searchApi");
+
+    var scrollId = searchResponse.getScrollId();
+    var searchHits = searchResponse.getHits().getHits();
+    consumer.accept(getInstanceIds(searchHits));
+
+    while (isNotEmpty(searchHits)) {
+      var scrollRequest = new SearchScrollRequest(scrollId).scroll(KEEP_ALIVE_INTERVAL);
+      var scrollResponse = performExceptionalOperation(
+        () -> elasticsearchClient.scroll(scrollRequest, DEFAULT), index, "scrollApi");
+      scrollId = scrollResponse.getScrollId();
+      searchHits = scrollResponse.getHits().getHits();
+      consumer.accept(getInstanceIds(searchHits));
+    }
```

Comment:
Would it work if replace both `consumer.accept(getInstanceIds(searchHits));` with just one right after line 78?